klaviyo/data_privacy.py

```diff
@@ -0,0 +1,39 @@
+from .api_helper import KlaviyoAPI
+from .exceptions import KlaviyoException
+
+from enum import Enum
+
+class AllowedIdTypes(Enum):
+    EMAIL = 'email'
+    PHONE_NUMBER = 'phone_number'
+    PERSON_ID = 'person_id'
+
+
+class DataPrivacy(KlaviyoAPI):
+    DATA_PRIVACY = 'data-privacy'
+    DELETION_REQUEST = 'deletion-request'
+
+    def request_profile_deletion(self, identifier, id_type='email'):
+        """Request a data privacy-compliant deletion for the person record corresponding to an email address,
+        phone number, or person identifier. If multiple person records exist for the provided identifier,
+        only one of them will be deleted.
+
+        https://www.klaviyo.com/docs/api/v2/data-privacy#post-deletion-request
+
+        Args:
+            identifier (str): The email address, phone number in E.164 format, or person_id corresponding to a person record to delete.
+            id_type (str): The type of identifier provided, can be one of: 'email', 'phone_number', or 'person_id'.
+        Returns:
+            (KlaviyoAPIResponse): Object with HTTP response code and data.
+        Raises:
+            (KlaviyoAPIException): Raised if invalid id_type is provided.
+        """
+        if id_type not in AllowedIdTypes:
```

Comment:
You're going to need to instantiate the enum first and the beauty of the enum is there's no need to check if it's included because a ValueError is raised if it's not a valid type. If we want to raise our own exception we could throw a try/except around that.
```
In [1]: from enum import Enum

In [2]: class AllowedIdTypes(Enum):
   ...:     EMAIL = 'email'
   ...:     PHONE_NUMBER = 'phone_number'
   ...:     PERSON_ID = 'person_id'
   ...:

In [3]: 'email' in AllowedIdTypes
Out[3]: False

In [4]: test = AllowedIdTypes('email')

In [5]: test
Out[5]: <AllowedIdTypes.EMAIL: 'email'>

In [6]: test in AllowedIdTypes
Out[6]: True

In [7]: bad = AllowedIdTypes('emailzzz')
ValueError: emailzzz is not a valid AllowedIdTypes
```

Reply:
My only concern on this approach is that the value error does not inform the developer using the SDK of what the allowed ID types are. How do we feel about something like this?
```python
try:
    AllowedIdTypes(id_type)
except ValueError:
    raise KlaviyoException('Invalid id_type provided, must be one of: {}'.format([t.value for t in AllowedIdTypes]))
```

Reply:
I think that's fine.